### PR TITLE
Drop `UseGardenerNodeAgent` feature gate, `cloud-config-{downloader,executor}`, and `oscommon` package 🧹

### LINKED ISSUE #4673: Remove usage of `docker` in Gardener, such that Gardener can use OS images without docker daemon installed
**How to categorize this issue?**

/area control-plane
/kind enhancement
/priority 3

**What would you like to be added**:
`docker` is still used in Gardener internals:
* The `systemd` unit responsible for downloading the cloud-config depends on the `docker` systemd unit: https://github.com/gardener/gardener/blob/2788f29310bb23faa224a72bbf946f12136e0f5f/pkg/operation/botanist/component/extensions/operatingsystemconfig/downloader/downloader.go#L138-L140
  * This probably makes sense, as the unit runs the script below, which uses `docker`
* Gardener preloads the hypercube image and uses `docker` to extract the binaries for kubelet and kubectl: https://github.com/gardener/gardener/blob/2788f29310bb23faa224a72bbf946f12136e0f5f/pkg/operation/botanist/component/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh#L80-L106
* Promtail uses a similar mechanism like above to extract the binary from an image: https://github.com/gardener/gardener/blob/2788f29310bb23faa224a72bbf946f12136e0f5f/pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/promtail/component.go#L60
* The root certificate updater restarts the docker daemon after updating the certs: https://github.com/gardener/gardener/blob/2788f29310bb23faa224a72bbf946f12136e0f5f/pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/rootcertificates/component.go#L62
* Gardener tunes some kernel parameters for docker-specific reasons: https://github.com/gardener/gardener/blob/2788f29310bb23faa224a72bbf946f12136e0f5f/pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kernelconfig/component.go#L89-L97

We should remove any usage of `docker` in Gardener, such that Gardener can use OS images without docker daemon installed.

**Proposal**:
* For downloading/preloading images and extracting binaries, we can replace `docker` with an alternative CLI which uses `containerd` instead: `ctr ` should be installed on the OSes supporting `containerd` already today. Alternatively, there is `nerdctl`, which offers a UX similar to `docker`, but supports additional stuff like lazy-pulling (which might come in handy specifically for extracting single binaries from an image: https://github.com/containerd/nerdctl/blob/master/docs/stargz.md)
* Drop the daemon reload after updating root certificates
* Investigate if kernel parameter tuning is still necessary/valid when using `containerd`. We might be able to remove this entirely or need very different optimizations

**Additional considerations**:
Before creating an OS image without support for `docker`, there's a couple of other things to consider:
* Remove the migration step in the CloudProfile controller, adding `docker` into the list of supported runtimes for each image: https://github.com/gardener/gardener/blob/a82aed23505004b507796b9f47a12f19b98d8fe6/pkg/controllermanager/controller/cloudprofile/cloudprofile_control.go#L122-L132
* Remove the validation in https://github.com/gardener/gardener/blob/6f878c51682ba75f8c6281b9f68d5a37998b2b8d/pkg/apis/core/validation/cloudprofile.go#L311-L313, added with https://github.com/gardener/gardener/pull/6480. 
* TODO: more considerations here

**Why is this needed**:
While it is possible for users to specify `containerd` to run their workload containers, Gardener still requires the docker daemon to be present and working in the OS images used for the worker nodes.

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area open-source
/kind cleanup

**What this PR does / why we need it**:
- This PR drops the graduated `UseGardenerNodeAgent` feature gate as well as all migration code related to it.
- This allows dropping `cloud-config-downloader` and `cloud-config-executor` bash scripts.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/8023
Fixes https://github.com/gardener/gardener/issues/4673

**Special notes for your reviewer**:
/cc @oliver-goetz @ScheererJ
🥂 kudos and many thanks to everybody involved, cc @majst01 @Gerrit91 @timebertt @vknabel and others 🍻 

**Release note**:

```breaking operator
The graduated `UseGardenerNodeAgent` feature gate has been dropped. Before upgrading to this Gardener version, make sure to have up-to-date OS extensions installed on your landscape: `os-gardenlinux@v0.24+`, `os-ubuntu@v1.25+`, `os-suse-chost@v1.25+`, `os-coreos@v1.20+`, `os-metal@v0.8+`, `provider-aws@v1.53.2+`.
```
```breaking developer
The deprecated `oscommon` package has been removed.
```
```noteworthy developer
The following fields in the`extensions.gardener.cloud/v1alpha1.OperatingSystemConfig` API have been deprecated and will be removed in a future version: `.spec.reloadConfigFilePath`, `.status.command`, `.status.units`, `.status.files`.
```

### CODE REVIEW

**Comment by timebertt** [comment]
There should be a 🍻 reaction for PRs like this.
Thanks, everybody. Well done! 👏

**Comment by majst01** [comment]
> There should be a 🍻 reaction for PRs like this. Thanks, everybody. Well done! 👏

We can 🍻  at the Hackathon then :-)

File: pkg/component/extensions/operatingsystemconfig/operatingsystemconfig_test.go [hunk 1]
```diff
@@ -311,30 +274,13 @@ var _ = Describe("OperatingSystemConfig", func() {
...
+						Purpose:   extensionsv1alpha1.OperatingSystemConfigPurposeReconcile,
+						CRIConfig: criConfig,
+						Units:     originalUnits,
+						Files:     originalFiles,
 					},
 				}
 
-				if useGardenerNodeAgent {
-					oscOriginal.Spec.Units = append(oscOriginal.Spec.Units, initUnits...)
-					oscOriginal.Spec.Files = append(oscOriginal.Spec.Files, extensionsv1alpha1.File{Path: "/var/lib/gardener-node-agent/init.sh"})
```
Comment (ScheererJ) [inline pkg/component/extensions/operatingsystemconfig/operatingsystemconfig_test.go:324]: Should this not still be executed? Maybe I overlooked it.

**Review by ScheererJ — COMMENTED** [review]
Awesome change. Only found a small thing. Looks good otherwise.

File: pkg/component/extensions/operatingsystemconfig/operatingsystemconfig_test.go (hunk 1 above)
Comment (rfranzke) [reply pkg/component/extensions/operatingsystemconfig/operatingsystemconfig_test.go:324]: As discussed in person, this change is done intentionally as we do not need to reconcile the `gardener-node-init` service once GNA is bootstrapped on a machine.

**Review by ScheererJ — COMMENTED** [review]
/lgtm

**Review by oliver-goetz — APPROVED** [review]
Farewell `bash` 🧹 
-4875 lines...nice cleanup thanks 🚀

**Comment by timebertt** [comment]
Great release note. Love the list of extension version requirements!

**Pull request merged**